openstack/nova/templates/cell2-conductor-deployment.yaml

```diff
@@ -29,29 +29,42 @@ spec:
         alert-service: nova
 {{ tuple . "nova" "conductor" | include "helm-toolkit.snippets.kubernetes_metadata_labels" | indent 8 }}
       annotations:
-        {{- if .Values.cell2.conductor.config_file.DEFAULT.statsd_enabled }}
+        {{- if or .Values.cell2.conductor.config_file.DEFAULT.statsd_enabled .Values.proxysql.mode }}
         prometheus.io/scrape: "true"
         prometheus.io/targets: {{ required ".Values.alerts.prometheus missing" .Values.alerts.prometheus | quote }}
         {{- end }}
         configmap-etc-hash: {{ include (print $.Template.BasePath "/etc-configmap.yaml") . | sha256sum }}
     spec:
 {{ tuple . "nova" "conductor" | include "kubernetes_pod_anti_affinity" | indent 6 }}
+{{ include "utils.proxysql.pod_settings" . | indent 6 }}
       terminationGracePeriodSeconds: {{ .Values.defaults.default.graceful_shutdown_timeout | add 5 }}
       hostname: nova-conductor
-      containers:
-        - name: nova-conductor
+      initContainers:
+        - name: dependencies
           image: {{ required ".Values.global.registry is missing" .Values.global.registry}}/ubuntu-source-nova-conductor:{{.Values.imageVersionNovaConductor | default .Values.imageVersionNova | default .Values.imageVersion | required "Please set nova.imageVersion or similar"}}
           imagePullPolicy: IfNotPresent
           command:
-            - dumb-init
             - kubernetes-entrypoint
           env:
             - name: COMMAND
-              value: "nova-conductor --config-file etc/nova/nova.conf --config-file /etc/nova/nova-conductor.conf --config-file /etc/nova/nova-cell2.conf"
+              value: "true"
```

Comment:
Why is the cell2 conductor so different from the cell1 conductor in that it has an init-container for waiting for the dependencies instead of in the "real" container and no `dumb-init` anymore for the conductor itself?

Reply:
Could you please explain why we have that init-container to wait for the dependencies instead of waiting in the real container? It seems like nova-conductor is the only Deployment doing that now. What's the adavantage in doing it in the init-container and why do others not need it?

Reply:
Thank you.